src/manhole/__init__.py

```diff
@@ -296,7 +300,10 @@ def handle_connection_repl(client):
     if _MANHOLE.redirect_stderr:
         patches.append(('w', ('stderr', '__stderr__')))
     try:
-        client_fd = client.fileno()
+        if hasattr(client, 'detach'):
+            client_fd = client.detach()
+        else:
+            client_fd = _ORIGINAL_DUP(client.fileno())
```

Comment:
When that may happen?

Should it use `fileno` at all in this case when it leads to known bugs?

Reply:
That's what the dup is for, there would be two numbers for the have underlying resource. So there won't be a double close since there would be two different numbers.

Reply:
Ah, true, didn't noticed that. Thanks